.github/workflows/rust_binaries_release.yml

```diff
@@ -0,0 +1,215 @@
+        if: matrix.target == 'aarch64-unknown-linux-gnu'
+        run: |
+          wget https://ziglang.org/download/0.11.0/zig-linux-x86_64-0.11.0.tar.xz
+          tar -xf zig-linux-x86_64-0.11.0.tar.xz
+          echo "ZIG_PATH=$PWD/zig-linux-x86_64-0.11.0/zig" >> $GITHUB_ENV
+          echo "$PWD/zig-linux-x86_64-0.11.0" >> $GITHUB_PATH
+          zig-linux-x86_64-0.11.0/zig version
+          echo "CARGO_ZIGBUILD_ZIG=$PWD/zig-linux-x86_64-0.11.0/zig" >> $GITHUB_ENV
+          cargo install cargo-zigbuild
+
+      - name: Build binaries
+        if: matrix.target == 'aarch64-unknown-linux-gnu'
+        env:
+          C_INCLUDE_PATH: /usr/lib/gcc-cross/aarch64-linux-gnu/11/include
+          CXX_INCLUDE_PATH: /usr/lib/gcc-cross/aarch64-linux-gnu/11/include/c++
+          OPENSSL_DIR: ${{ env.OPENSSL_DIR }}
+          OPENSSL_LIB_DIR: ${{ env.OPENSSL_LIB_DIR }}
+          OPENSSL_INCLUDE_DIR: ${{ env.OPENSSL_INCLUDE_DIR }}
+          PKG_CONFIG_PATH: ${{ env.PKG_CONFIG_PATH }}
+          PKG_CONFIG_ALLOW_CROSS: ${{ env.PKG_CONFIG_ALLOW_CROSS }}
+          PKG_CONFIG_SYSROOT_DIR: ${{ env.PKG_CONFIG_SYSROOT_DIR }}
+          OPENSSL_STATIC: ${{ env.OPENSSL_STATIC }}
+          RUSTFLAGS: '-C link-arg=-lstdc++ -C link-arg=-lpthread -C link-arg=-lc'
+        run: cargo zigbuild ${{ steps.cargo_args.outputs.args }}
+
+      - name: Build binaries
+        if: matrix.target != 'aarch64-unknown-linux-gnu'
+        run: cargo build ${{ steps.cargo_args.outputs.args }}
+
+      - name: Compress artifacts using gzip
+        run: |
+          mkdir -p artifacts
+          echo '${{ needs.binary_matrix.outputs.binaries }}' | jq -r '.[]' | while read binary; do
+            tar -czf artifacts/"$binary"_${{ matrix.target }}.tar.gz -C target/${{ matrix.target }}/release "$binary"
+          done
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifacts-${{ matrix.target }}
+          path: artifacts/*
+          retention-days: 2
+
+  release:
+    runs-on: ubuntu-latest
+    needs: [binary_matrix, build]
+    strategy:
+      matrix:
+        binary: ${{ fromJSON(needs.binary_matrix.outputs.binaries) }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Download Artifact
+        uses: actions/download-artifact@v4
+        with:
+          path: artifacts/
+          merge-multiple: true
+
+      - name: Get version info
+        id: version_info
+        shell: bash
+        run: |
+          echo "target_commit=${{ github.sha }}" >> $GITHUB_OUTPUT
+
+          if [ "${{ github.ref }}" == "refs/heads/master" ]; then
+            version=$(cargo metadata --format-version 1 --no-deps | jq -r '.packages[] | select(.name == "${{ matrix.binary }}") | .version')
+            version="${{ matrix.binary }}-$version"
+            echo "Master version: $version"
+
+            if gh release view "v$VERSION" >/dev/null 2>&1; then
```

Comment:
`$VERSION` here is capitalized?

and the version won't be `v` prefixed, right?

at least according to the snippet above

```shell
version="${{ matrix.binary }}-$version"
```

Here's how actix does it: https://github.com/actix/actix-web/releases